### PR TITLE
Multiple man-pages: Move to appropriate section

### PR DESCRIPTION
**Pull request by debdrup**
### Motivation and Context
These manual pages belong in section 7 of the manual pages as that
section, according to intro(7) on both FreeBSD and Linux, is meant to
contain miscellaneous information pages whereas section 5 describes file
formats (a good example being vdev_id.conf in zfs/man/man5/).

While here, an errant space has also been removed.

Signed-off-by:	Daniel Ebdrup Jensen	<debdrup@FreeBSD.org>

### How Has This Been Tested?
Mandoc has been run, but the change doesn't change a whole lot of roff formatting.

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [X] Documentation (a change to man pages or other documentation)

### Checklist:
- [ ] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [X] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: man/man8/zpool-get.8 [hunk 1]
```diff
@@ -99,4 +99,4 @@ values.
 .Sh SEE ALSO
 .Xr zpoolprops 8 ,
```
Comment (freqlabs) [inline man/man8/zpool-get.8:100]:
```suggestion
.Xr zpoolprops 7 ,
```

File: man/man8/zpool-upgrade.8 [hunk 2]
```diff
@@ -102,7 +102,7 @@ supported legacy version number.
 .El
 .El
 .Sh SEE ALSO
-.Xr zpool-features 5 ,
+.Xr zpool-features 7 ,
 .Xr zpoolconcepts 8 ,
 .Xr zpoolprops 8 ,
```
Comment (freqlabs) [inline man/man8/zpool-upgrade.8:107]:
```suggestion
.Xr zpoolprops 7 ,
```

Comment (freqlabs) [inline man/man8/zpool-upgrade.8:107]:
```suggestion
.Xr zpoolconcepts 7 ,
.Xr zpoolprops 7 ,
```

**Comment by freqlabs** [comment]
I spotted a few missed cross references to the moved pages just in the diff context, so let's make sure to grep and get all of them.

**Comment by debdrup** [comment]
> I spotted a few missed cross references to the moved pages just in the diff context, so let's make sure to grep and get all of them.

You're right, I missed quite a few.
Apparently my own grep wasn't thorough enough.
I'll fix 'em. :)

While here, I also noticed that I forgot some other things, including bumping .Dd - so I'll take care of that too.

**Comment by freqlabs** [comment]
Thanks. There are still 3 files renamed without changes, so they still have the old section number in them:
* spl-module-parameters
* zfs-module-parameters
* zfs-events

**Comment by debdrup** [comment]
> Thanks. There are still 3 files renamed without changes, so they still have the old section number in them:
> 
>     * spl-module-parameters
> 
>     * zfs-module-parameters
> 
>     * zfs-events

Ah, good catch!

I have to admit that I wasn't aware of the .ST macro.
It doesn't seem reasonable to assume it has anything to do with .St [as described here](http://manpages.bsd.lv/mdoc.html#x5c265374), nor is it mentioned in [mdoc(7)](https://man.freebsd.org/mdoc(7)), nor [roff(7)](https://man.freebsd.org/roff(7)), so I'm a bit confused about where it originates.

**Comment by freqlabs** [comment]
FYI there are a few other batches of manpage changes open right now such as #12125 and #12129 which may conflict a bit, but should also make the manpages much cleaner WRT mandoc style.

**Comment by nabijaczleweli** [comment]
Can we hold off on this until after #12125 and #12129 please? I touch all but two manpages (and rewrite some in parts or in full) in those and it was already a tremendous effort, and just rebasing that onto the new `ztest.1` was non-obvious.

If you like, if you compile the list of renames from here I can run those myself after these land (which will be likely simpler to get right since most manpages properly `.Xr` now) and submit that as a separate PR.

**Comment by debdrup** [comment]
> Yeah, can we hold off on this until after #12125 and #12129 please? I touch all but two manpages (and rewrite some in parts or in full) in those and it was already a _tremendous_ effort (and I don't say that lightly), and just rebasing that onto the new `ztest.1` was non-obvious.

Yeah, I can imagine, as that's quite a bit of work.

> If you like, if you compile the list of renames from here I can run them myself after these land (which will be likely simpler to get right since most manpages properly `.Xr` now) and submit that?

Sure, I don't mind as long as it gets done, that's what I care about. :)

The files are:
* zpool-features.5
* spl-module-parameters.5
* zfs-module-parameters.5
* zfs-events.5
* zfsconcepts.8
* zfsprops.8
* zpoolconcepts.8
* zpoolprops.8

This can also be seen by looking the subtractions in man5/Makefile.am and man8/Makefile.am [here](https://github.com/openzfs/zfs/pull/12149/files).

Can I get a Co-authored-by: Daniel Ebdrup Jensen <debdrup@FreeBSD.org>?

**Comment by nabijaczleweli** [comment]
Sure thing

**Comment by behlendorf** [comment]
@debdrup it'd be great to get your thoughts on the right location for some of these man pages.  @nabijaczleweli posted a nice detailed https://github.com/openzfs/zfs/pull/12196#issuecomment-856281510 which goes over rationals for various locations.

**Pull request closed without merging**